src/StatusBar.ts

```diff
@@ -1,72 +1,254 @@
+  }
+
+  private request(source: string, status: Status, details?: string) {
+    const request: StatusUpdateRequest = {
       source,
       status,
       details,
-    })
-    this.updateStatus()
+    }
+    this.requests.set(source, request)
+    this.updateStatus(request)
   }
-  private updateStatus() {
-    for (const status of this.priorities) {
-      const queueItem = this.queue.find(item => item.status === status)
-      if (queueItem) {
-        this.render(queueItem)
-        break
+  private updateStatus(request: StatusUpdateRequest) {
+    this.updateActiveStatus(request)
+    this.updateSummaryStatus()
+  }
+
+  private get activeFolder() {
+    if (!this._activeFolder && vscode.window.activeTextEditor) {
+      const folder = vscode.workspace.getWorkspaceFolder(vscode.window.activeTextEditor.document.uri)
+      if (folder) {
+        this._activeFolder = folder.name
+      }
+    }
+    return this._activeFolder
+  }
+
+  private updateActiveStatus(request?: StatusUpdateRequest) {
+    if (request && this.activeFolder) {
+      if (request.source === this.activeFolder) {
+        this.render(request, this.activeStatusItem)
       }
+      return
+    }
+
+    // find the active item from requests
+    let _request = null
+    if (this.activeFolder) {
+      _request = this.requests.get(this.activeFolder)
+    }
+    if (!_request && this.requests.size === 1) {
+      _request = this.requests.values().next().value
+    }
+
+    if (_request) {
+      this.render(_request, this.activeStatusItem)
+    } else {
+      this.activeStatusItem.hide()
     }
   }
-  private render(queueItem: QueueItem) {
-    clearInterval(this.statusBarSpinner)
-    const message = this.getMessageByStatus(queueItem.status)
-    if (queueItem.status === 'running') {
-      this.statusBarSpinner = setInterval(() => this.render(queueItem), 100)
+
+  private updateSummaryStatus() {
+    if (this.needsSummaryStatus()) {
+      this.updateSummaryOutput()
+
+      let summaryStatus: StatusUpdateRequest | undefined
+      let prev = 99
+      for (const r of this.requests.values()) {
+        const idx = this.priorities.indexOf(r.status)
+        if (idx >= 0 && idx < prev) {
+          summaryStatus = r
+          prev = idx
+        }
+      }
+
+      if (summaryStatus) {
+        this.render(summaryStatus, this.summaryStatusItem)
+        return
+      }
     }
-    this.statusBarItem.text = `${this.statusKey} ${message} ${queueItem.details || ''}`
+    this.summaryStatusItem.hide()
   }
+
+  private render(request: StatusUpdateRequest, statusBarItem: SpinnableStatusBarItem) {
+    statusBarItem.clearSpinner()
+
+    const message = this.getMessageByStatus(request.status)
+    if (request.status === 'running') {
+      statusBarItem.startSpinner(() => this.render(request, statusBarItem))
+    }
+
+    switch (statusBarItem.type) {
+      case StatusType.active:
+        const details = !this.needsSummaryStatus() && request.details ? request.details : ''

```

Comment:
I didn't see any lint rules/best-practice regarding the condition in ternary expression, however, did find the [multiline-ternary](https://eslint.org/docs/rules/multiline-ternary) in eslint, which defaults to separate lines. I could change it accordingly, and it will probably make the condition more readable.

Reply:
`const details = (!this.needsSummaryStatus() && request.details) || ''` probably wouldn't be obvious enough?
Otherwise I'm also fine with leaving it as it is.